src/app/business/services/dynamic-form/dynamic-form.component.ts

```diff
@@ -0,0 +1,196 @@
+  options = {
+    headers: {
+        'X-Auth-Token': localStorage['auth-token']
+    } 
+  };
+  
+
+  constructor(private router: Router,
+    private ActivatedRoute:ActivatedRoute,
+    public i18n: I18NService,
+    private wfservice: WorkflowService,
+    private profileService: ProfileService,
+    private paramStor: ParamStorService,
+    private fb: FormBuilder) {
+      this.default_parameters['auth_token'] = localStorage['auth-token'];
+      this.default_parameters['user_id'] = this.default_parameters['userId'] = this.paramStor.CURRENT_USER().split("|")[1];
+      this.default_parameters['tenant_id'] = this.paramStor.CURRENT_TENANT().split("|")[1];
+    }
+
+  ngOnInit() {
+    let self = this;
+
+    this.dataObject['instanceName'] = {
+      "description": "Name of the Instance",
+      "required": true,
+      "type": "string",
+      "validation": {
+        "required": true
+      },
+      "instanceInfo" : true
+    };
+
+    this.dataObject['instanceDescription'] = {
+      "description": "Description of the Instance",
+      "required": true,
+      "type": "string",
+      "validation": {
+        "required": true
+      },
+      "instanceInfo" : true
+    };
+    // remap the API to be suitable for iterating over it
+    this.objectProps =
+      Object.keys(this.dataObject)
+        .map(prop => {
+          return Object.assign({}, { key: prop} , this.dataObject[prop]);
+        });
+      
+      // setup the form
+      const formGroup = {};
+
+      _.each(this.objectProps, function(item){
+        item['showThis'] = true;
+        item['label'] = item['key'].replace(/_/g, ' ').replace(/(?: |\b)(\w)/g, function(key, p1) {
+              return key.toUpperCase();    
+        });
+        if(item['key']=='timeout'){
+          item['required'] = false;
+        }
+        if(item['type']=='object' || item['required'] == false || !_.has(item, 'required')){
+          item['showThis'] = false;
+        }
+        //Check if the key is a default parameter and can be passed from code. User need not enter.
+        if(item['required'] == true){
+          item['validation'] = {required: true};
+          if(_.has(self.default_parameters, item['key'])){
+            item['showThis'] = false;
+            item['default'] = item['value'] = self.default_parameters[item['key']];
+          }
+        }
+
+        if(item['key']=='profile_id' && item['showThis']){
+          self.getProfiles();
+          item['inputType'] = "select";
+          item['options'] = self.profileOptions;
+        }
+        
+        if(item['type'] == "string" && item['key'] != 'profile_id'){
+          item['inputType'] = "text";
+        } else if(item['type'] == "boolean"){
+          item['inputType'] = "radio";
+          item['options'] = [
+            { label: "True", value: 'true'},
+            { label: "False", value: 'false'}
+          ];
+        } else if(item['type'] == "integer"){
+          item['inputType'] = "number";
+        }
```

Comment:
If the parameter type is different from string, boolean, integers like number, object or array. Then what will be the inputType?

For parameter input type:
https://docs.stackstorm.com/actions.html

Reply:
Since this information was not available at the time I've only considered the input types that were provided viz. string, boolean, integers.
For type object and and array parsers will have to be written and in the front end editors will be provided as key value pairs. 
This will be taken care in future. Please add this as an issue in the dashboard.

Reply:
Please track this as an issue.